src/methods.jl

```diff
@@ -10,318 +19,453 @@ derivatives.
-            _check_p_q(p, q)
-            grid = $gridf(p)
-            coefs = _coefs(grid, p, q)
-            hist = History(; adapt=adapt, kwargs...)
-            return $D{typeof(grid), typeof(coefs)}(Int(p), Int(q), grid, coefs, hist)
-        end
+julia> fdm(sin, 1) - cos(1)  # Check the error.
+-2.4313884239290928e-14
 
-        @doc """
-            FiniteDifferences.$($(Meta.quot(D)))(p, q; kwargs...)
-            $($(Meta.quot(fdmf)))(p, q; kwargs...)
-
-        Construct a $($lcname) finite difference method of order `p` to compute the `q`th
-        derivative.
-        See [`FiniteDifferenceMethod`](@ref) for more details.
-        """
-        ($D, $fdmf)
-    end
+julia> FiniteDifferences.estimate_step(fdm, sin, 1.0)  # Computes step size and estimates the error.
+(0.0010632902144695163, 1.9577610541734626e-13)
+```
+"""
+function (m::FiniteDifferenceMethod)(
+    f::Function,
+    x::T;
+    factor::Int=1,
+    max_step=convert(T, 0.1)
+) where T<:AbstractFloat
+    # The automatic step size calculation fails if `m.q == 0`, so handle that edge case.
+    iszero(m.q) && return f(x)
+    h, _ = estimate_step(m, f, x, factor=factor, max_step=max_step)
+    return m(f, x, h)
+end
+# Handle arguments that are not floats. Assume that converting to float is desired.
+function (m::FiniteDifferenceMethod)(f::Function, x::T; kw_args...) where T<:Real
+    return m(f, float(x); kw_args...)
 end
 
 """
-    FiniteDifferences.Nonstandard(grid, q; kwargs...)
+    (m::FiniteDifferenceMethod)(
+        f::Function,
+        x::T,
+        h
+    ) where T<:AbstractFloat
+
+Estimate the derivative of `f` at `x` using the finite differencing method `m` and a given
+step size.
+
+# Arguments
+- `f::Function`: Function to estimate derivative of.
+- `x::T`: Input to estimate derivative at.
+- `h`: Step size.
 
-An finite differencing method which is constructed based on a user-defined grid. It is
-nonstandard in the sense that it represents neither forward, backward, nor central
-differencing.
-See [`FiniteDifferenceMethod`](@ref) for further details.
+# Returns
+- Estimate of the derivative.
+
+# Examples
+
+```julia-repl
+julia> fdm = central_fdm(5, 1)
+FiniteDifferenceMethod:
+  order of method:       5
+  order of derivative:   1
+  grid:                  [-2, -1, 0, 1, 2]
+  coefficients:          [0.08333333333333333, -0.6666666666666666, 0.0, 0.6666666666666666, -0.08333333333333333]
+
+julia> fdm(sin, 1, 1e-3)
+0.5403023058679624
+
+julia> fdm(sin, 1, 1e-3) - cos(1)  # Check the error.
+-1.7741363933510002e-13
+```
 """
-function Nonstandard(grid::AbstractVector{<:Real}, q::Integer; adapt=0, kwargs...)
-    p = length(grid)
-    _check_p_q(p, q)
-    coefs = _coefs(grid, p, q)
-    hist = History(; adapt=adapt, kwargs...)
-    return Nonstandard{typeof(grid), typeof(coefs)}(Int(p), Int(q), grid, coefs, hist)
+function (m::FiniteDifferenceMethod)(
+    f::Function,
+    x::T,
+    h
+) where T<:AbstractFloat
```

Comment:
This all fits on one line under the 92 caracter limit.
```suggestion
function (m::FiniteDifferenceMethod)(f::Function, x::T, h) where T<:AbstractFloat
```